packages/Datepicker/src/Datepicker.tsx

```diff
@@ -1,193 +1,220 @@
+                className={classes}
+                value={formatDate(selectedDay) || null}
+                onChange={handleChange}
+                isDisabled={disabled}
+                highlightToday={highlightToday}
+                isDateUnavailable={isDateUnavailable}
+                minValue={formatDate(minDate)}
+                maxValue={formatDate(maxDate)}
+            />
+            {isClearable && clearLabel && (
+                <Button
+                    onClick={handleClear}
+                    className="ids-datepicker__action"
+                    appearance={{ type: "ghost", variant: "danger" }}
+                >
+                    {clearLabel}
+                </Button>
+            )}
+        </>
     );
 
-    // the calendar return an object with utc and local
-    if (onChange) {
-      onChange({
-        utc: local.toAbsoluteString(),
-        local: local.toString(),
-      });
-    }
-  };
-
-  const handleClear = () => {
-    if (onClear) {
-      onClear();
+    const inputProps: InputProps = {
+        placeholder,
+        disabled,
+        error,
+        value,
+        prefixIcon: <IconCalendar />,
+        type: "text",
+        onFocus,
+        onChange: handleInputChange
+    };
+
+    if (readOnly) {
+        inputProps.readOnly = readOnly;
     }
 
-    if (onChange) {
-      onChange(null);
-    }
-  };
-
-  const isDateUnavailable = (date: DateValue) => {
-    if (weekendUnavailable && locale) {
-      return isWeekend(date, locale);
-    }
-
-    return false;
-  };
-
-  const classes = cx('ids-datepicker', {
-    'ids-datepicker--disabled': disabled,
-  });
-
-  const calendar = (
-    <>
-      <Calendar
-        aria-label={ariaLabel}
-        className={classes}
-        value={formatDate(selectedDay)}
-        onChange={handleChange}
-        isDisabled={disabled}
-        highlightToday={highlightToday}
-        isDateUnavailable={isDateUnavailable}
-        minValue={formatDate(minDate)}
-        maxValue={formatDate(maxDate)}
-      />
-      {isClearable && clearLabel && (
-        <Button
-          onClick={handleClear}
-          className="ids-datepicker__action"
-          appearance={{ type: 'ghost', variant: 'danger' }}
-        >
-          {clearLabel}
-        </Button>
-      )}
-    </>
-  );
-
-  return (
-    <Dropdown
-      isOpen={!disabled && isOpen}
-      onClose={onClose}
-      content={calendar}
-      size="medium"
-      dataTest={dataTest}
-      {...rest}
-    >
-      <Input
-        type="text"
-        disabled={disabled}
-        error={error}
-        value={value}
-        placeholder={placeholder}
-        prefixIcon={<IconCalendar />}
-        onFocus={onFocus}
-      />
-    </Dropdown>
-  );
+    return (
+        <I18nProvider locale={locale}>
+            <Dropdown
+                isOpen={!disabled && isOpen}
+                onClose={onClose}
+                content={calendar}
+                size="medium"
```

Comment:
what's the default dropdown size? is it medium?

Reply:
the default is xsmall, so this is still valid. Keep in mind, some of these files look like they have a lot of changes because when I save them, it updates with the new formatting rules we added recently to be like hopper.